pkg/ssh/ssh.go

```diff
@@ -179,20 +189,8 @@ const sshDialTimeout = 150 * time.Second
 var realTimeoutDialer sshDialer = &timeoutDialer{&realSSHDialer{}, sshDialTimeout}
 
 func (d *timeoutDialer) Dial(network, addr string, config *ssh.ClientConfig) (*ssh.Client, error) {
-	var client *ssh.Client
-	errCh := make(chan error, 1)
-	go func() {
-		defer runtime.HandleCrash()
-		var err error
-		client, err = d.dialer.Dial(network, addr, config)
-		errCh <- err
-	}()
-	select {
-	case err := <-errCh:
-		return client, err
-	case <-time.After(d.timeout):
```

Comment:
Wondering why that never interrupted the leaky routine?.. What could we learn from that anti-pattern? "Less (go func() + defer + time.After) == more kittens saved?"

A great solution to the nice bug with corner cases handling and `go func()`!